internals/secrethub/audit.go

```diff
@@ -1,41 +1,68 @@
 package secrethub
 
 import (
+	"bytes"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"io"
+	"os"
+	"os/exec"
 	"strings"
-	"text/tabwriter"
+
+	"golang.org/x/crypto/ssh/terminal"
+
+	"github.com/secrethub/secrethub-go/pkg/secrethub/iterator"
 
 	"github.com/secrethub/secrethub-cli/internals/cli/ui"
 	"github.com/secrethub/secrethub-cli/internals/secrethub/command"
 	"github.com/secrethub/secrethub-go/pkg/secrethub"
-	"github.com/secrethub/secrethub-go/pkg/secrethub/iterator"
 
 	"github.com/secrethub/secrethub-go/internals/api"
 )
 
+var (
+	errPagerNotFound = errors.New("no terminal pager available. Please configure a terminal pager by setting the $PAGER environment variable or install less or more")
+)
+
+const (
+	pagerEnvvar            = "$PAGER"
+	defaultTerminalWidth   = 80
+	fallbackPagerLineCount = 100
+)
+
 // AuditCommand is a command to audit a repo or a secret.
 type AuditCommand struct {
-	io            ui.IO
-	path          api.Path
-	useTimestamps bool
-	timeFormatter TimeFormatter
-	newClient     newClientFunc
-	perPage       int
+	io                 ui.IO
+	newPaginatedWriter func(io.Writer) (pager, error)
+	path               api.Path
+	useTimestamps      bool
+	timeFormatter      TimeFormatter
+	newClient          newClientFunc
+	terminalWidth      func(int) (int, error)
+	perPage            int
+	json               bool
 }
 
 // NewAuditCommand creates a new audit command.
 func NewAuditCommand(io ui.IO, newClient newClientFunc) *AuditCommand {
 	return &AuditCommand{
-		io:        io,
-		newClient: newClient,
+		io:                 io,
+		newPaginatedWriter: newPaginatedWriter,
+		newClient:          newClient,
+		terminalWidth: func(fd int) (int, error) {
+			w, _, err := terminal.GetSize(fd)
+			return w, err
+		},
 	}
 }
 
 // Register registers the command, arguments and flags on the provided Registerer.
 func (cmd *AuditCommand) Register(r command.Registerer) {
-	clause := r.Command("audit", "Show the audit log.")
+	clause := r.Command("audit", "Show the audit log.\nIf the output of the command is parsed by a script the --json flag must be used.")
```

Comment:
- I will add a new line
- This was done with the intent of signaling the user that the tabular output is not meant for scripting as it will not be kept backwards compatible. I'm not sure if putting it in the `--json` flag description will achieve the same result, so I'll rephrase this sentence.